### PR TITLE
File downloader tests

### PR DESCRIPTION
**Pull request by Mecharyry**
Add tests for the `NetworkFileDownloader`

### CODE REVIEW

File: library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        given(httpClient.execute(any(NetworkRequest.class))).willReturn(
+                aNetworkResponse().withInputStream(new ByteArrayInputStream(BYTES_TO_RECEIVE)).build()
+        );
+    }
+
+    @Test
+    public void createsRequestWithDownloadedBytesHeader_whenKnown() {
+        networkFileDownloader.startDownloading(ANY_RAW_URL, KNOWN_FILE_SIZE, callback);
+
+        verify(requestCreator).createDownloadRequestWithDownloadedBytesHeader(ANY_RAW_URL, KNOWN_FILE_SIZE.currentSize(), KNOWN_FILE_SIZE.totalSize());
```
Comment (zegnus) [inline library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java]: I think that this is testing an interaction with an internal collaborator. If we refactor this class and the way that the internal collaborators work we will then have to come here and change this again. We should only test what the public methods/callbacks return and interaction with external collaborators. What do you think? The idea is to allow internal refactoring without having to change tests, while keeping behaviour tested

Comment (Mecharyry) [reply library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java]: 👍 very true. Updated in 2703344

File: library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        networkFileDownloader.startDownloading(ANY_RAW_URL, KNOWN_FILE_SIZE, callback);
+
+        byte[] expectedBytes = new byte[4096];
+        expectedBytes[0] = "s".getBytes()[0];
+
+        verify(callback).onBytesRead(expectedBytes, 1);
+    }
+
+    @Test
+    public void closesByteStream_whenDownloading() throws IOException {
```
Comment (ataulm) [inline library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java]: should this be `closesByteStream_afterDownloaded` or something? beware the test names that describe the implementation over describing the test case

Comment (Mecharyry) [reply library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java]: a02ab82

**Pull request merged**